src/components/shared/Home/PaginatedPosts.tsx

```diff
@@ -0,0 +1,67 @@
+"use client";
+import React, { useState } from "react";
+import PostCard from "./PostCard";
+import { Note } from "~/types";
+
+import {
+  Pagination,
+  PaginationContent,
+  PaginationItem,
+  PaginationNext,
+  PaginationPrevious,
+} from "@/components/ui/pagination";
+
+interface PaginatedPostsProps {
+  posts: string;
+}
+
+const PaginatedPosts: React.FC<PaginatedPostsProps> = ({ posts }) => {
+  const postsPerPage = 3;
+  const allPosts = JSON.parse(posts) as Note[];
+  const totalPages = Math.ceil(allPosts.length / postsPerPage);
+  const [startIndex, setStartIndex] = useState(0);
+  const [endIndex, setEndIndex] = useState(postsPerPage);
+
+  return (
+    <React.Fragment>
+      {allPosts.slice(startIndex, endIndex).map((post) => {
+        return <PostCard key={post.id} post={post} isShort={false} />;
+      })}
+      <Pagination>
+        <PaginationContent>
+          <PaginationItem>
+            <PaginationPrevious
+              className={
+                startIndex === 0 ? "pointer-events-none opacity-50" : undefined
+              }
+              onClick={() => {
+                setStartIndex(startIndex - postsPerPage);
+                setEndIndex(endIndex - postsPerPage);
+              }}
```

Comment:
PaginationPrevious requires an 'href' prop. Let's see if we can rework this to use an href so that we can fix the type error.

![image](https://github.com/dalbanhi/git-note/assets/4108484/0a8e260d-840a-424f-a8ce-c8facb97432b)

If we're no longer using this file or the `RecentPostsList` file we can probably remove them instead of fixing and reworking typescript errors